Symfony/CS/Fixer/Symfony/SingleQuoteFixer.php

```diff
@@ -35,12 +35,13 @@ public function fix(\SplFileInfo $file, $content)
             if (
                 '"' === $content[0] &&
                 false === strpos($content, "'") &&
-                // regex: odd number of backslashes, not followed by double quote
-                !preg_match('/(?<!\\\\)(?:\\\\{2})*\\\\(?!["\\\\])/', $content, $m)
+                // regex: odd number of backslashes, not followed by dollar or double quote
+                !preg_match('/(?<!\\\\)(?:\\\\{2})*\\\\(?![("\$)\\\\])/', $content, $m)
```

Comment:
Oh, and it is not necessary to escape the dollar sign within character classes.
So, this should work: `["$\\\\]`